File: CarHomeWidget/src/main/java/com/anod/car/home/incar/ChangeBrightnessActivity.java
package com.anod.car.home.incar;

import android.app.Activity;
import android.os.Bundle;
import android.os.Handler;
import android.view.WindowManager;

public class ChangeBrightnessActivity extends Activity {
    final static String EXTRA_BRIGHT_LEVEL = "bright_level";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        float bt = getIntent().getFloatExtra(EXTRA_BRIGHT_LEVEL, 1.0f);

        WindowManager.LayoutParams lp = getWindow().getAttributes();
        lp.screenBrightness = bt;
        getWindow().setAttributes(lp);
    }

    @Override
    protected void onPostCreate(Bundle savedInstanceState) {
        super.onPostCreate(savedInstanceState);
        Handler handler = new Handler();
        handler.postDelayed(new Runnable() {
            @Override
            public void run() {
                ChangeBrightnessActivity.this.finish();
            }
        }, 500);
    }
}

File: CarHomeWidget/src/main/java/com/anod/car/home/incar/ChangeBrightnessActivity.kt
package com.anod.car.home.incar

import android.app.Activity
import android.os.Bundle
import android.os.Handler
import android.view.WindowManager

class ChangeBrightnessActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val bt = intent.getFloatExtra(EXTRA_BRIGHT_LEVEL, 1.0f)

        val lp = window.attributes
        lp.screenBrightness = bt
        window.attributes = lp
    }

    override fun onPostCreate(savedInstanceState: Bundle?) {
        super.onPostCreate(savedInstanceState)
        val handler = Handler()
        handler.postDelayed({ this@ChangeBrightnessActivity.finish() }, 500)
    }

    companion object {
        internal const val EXTRA_BRIGHT_LEVEL = "bright_level"
    }
}
